polkadot/runtime/src/utils.rs

```diff
@@ -40,7 +39,7 @@ pub fn inherent_extrinsics(data: ::primitives::InherentData, runtime_version: Ru
 		make_inherent(Call::Parachains(ParachainsCall::set_heads(data.parachain_heads))),
 	];
 
-	if !data.offline_indices.is_empty() && runtime_version.spec_version == 4 {
```

Comment:
ugh. this presents a potential difficulty. the `InherentData` that the older runtime is expecting has a different meaning to the `InherentData` that the newer runtime expects (and that this authoring code provides). in particular, the old field `offline_nodes` are the nodes that have missed any further block proposals at least 5 minutes after one missed proposal. the new field, of the same name, are the nodes that missed a block proposal in the last block. two validators running different versions (0.2.4 and 0.2.6) will not be able to find consensus on what that inherent which should go in is.

it'll probably be fine, since after 60s if the validators can't agree on the offline nodes, they don't bother including them at all. but it's not ideal.

Reply:
Validators running `0.2.4` won't be able to author after the runtime upgrade anyway. And in the future such breaking changes are supposed to bump `autoring_version`.